gantry/simulator/CMakeLists.txt

```diff
@@ -1,18 +1,67 @@
+# Load free for the posix port
+find_package(FreeRTOS)
+
 # Set up the include directories
 include_directories(
         .
+        ../firmware
         ${CMAKE_CURRENT_SOURCE_DIR}/../../include
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/include
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils
+
+)
 
+# Create the FreeRTOS source file list
+FILE(
+        GLOB FREERTOS_SOURCES
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/*.c
 )
 
+# Add the posix port files
+list(APPEND FREERTOS_SOURCES "${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/MemMang/heap_3.c")
+list(APPEND FREERTOS_SOURCES "${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils/wait_for_event.c")
+list(APPEND FREERTOS_SOURCES "${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix/port.c")
+
+
+# Create the freertos lib
+add_library(freertos-gantry STATIC ${FREERTOS_SOURCES})
+
+if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Linux")
+    add_compile_definitions(USE_SOCKETCAN)
+endif()
+
+add_compile_definitions(ENABLE_LOGGING)
+
+set(GANTRY_SIMULATOR_SRC
+        ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/interfaces.cpp
+        ../firmware/freertos_idle_timer_task.cpp
```

Comment:
same with the include - IMO we really shouldn't be mixing firmware and simulator code even at the expense of duplication.

Reply:
The amount of duplication in this codebase makes me pukey. But ok.